api/src/main/java/org/eclipse/microprofile/openapi/models/media/Schema.java

```diff
@@ -1192,4 +1201,928 @@ default Schema oneOf(List<Schema> oneOf) {
+     *
+     * @param example
+     *            the example to add to the examples list
+     * @return current Schema instance
+     * @since 4.0
+     */
+    Schema addExample(Object example);
+
+    /**
+     * Removes an example from the examples list.
+     *
+     * @param example
+     *            the example to remove from the examples list
+     * @since 4.0
+     */
+    void removeExample(Object example);
+
+    /**
+     * Gets a schema property by name.
+     * <p>
+     * Allows access to arbitrary properties in a schema object, allowing use of alternative schema dialects which use
+     * different property names (or the same property names with different data types).
+     * <p>
+     * When using the standard schema dialect, this method can be used to retrieve values set by other methods. E.g.
+     *
+     * <pre>
+     * {@code
+     * schema.setMinimum(new BigDecimal(3));
+     * BigDecimal minimum = (BigDecimal) schema.get("minimum"); // returns 3
+     * }
+     * </pre>
+     *
+     * @param propertyName
+     *            the property name
+     * @return the value of the named property, or {@code null} if a property with the given name is not set
+     */
+    Object get(String propertyName);
+
+    /**
+     * Sets a schema property.
+     * <p>
+     * Allows the modifications of arbitrary schema properties in a schema properties, allowing use of alternative
+     * schema dialects which use different property names (or the same property names with different data types).
+     * <p>
+     * Passing {@code null} as the {@code value} removes the property from the schema object.
+     * <p>
+     * {@code value} must be one of the following types, otherwise non-portable behavior results:
+     * <ul>
+     * <li>Any primitive type
+     * <li>Any primitive wrapper class
+     * <li>{@code null}
+     * <li>{@code String}
+     * <li>{@code BigDecimal}
+     * <li>{@code BigInteger}
+     * <li>Any type which {@link OASFactory} can create
+     * <li>Any Enumeration
+     * <li>{@code List<T>} where {@code T} is a permitted type
+     * <li>{@code Map<String, T>} where {@code T} is a permitted type
+     * </ul>
+     *
+     * <p>
+     * When using the standard schema dialect, values set by this method can be retrieved by other methods. E.g.
+     *
+     * <pre>
+     * {@code
+     * schema.set("minimum", new BigDecimal(3));
+     * BigDecimal minimum = schema.getMinimum(); // returns 3
+     * }
+     * </pre>
+     *
+     * @param propertyName
+     *            the property name
+     * @param the
+     *            value to set
+     */
+    void set(String propertyName, Object value);
+
+    /**
+     * Gets all properties of a schema.
+     * <p>
+     * Equivalent to calling {@link #get(String)} for each property and putting them all into a {@code Map}.
+     *
+     * @return a {@code Map} of property names to their corresponding values
+     */
+    Map<String, ?> getAll();
```

Comment:
It would return only those that have been set, matching what is output when the object is serialized.

I hadn't really considered whether we should support keys with a `null` value in the output. The `set` method currently doesn't support setting a `null` value and I haven't seen anything that would need that support in the JSON Schema spec.

Reply:
I've pushed a commit to clarify this

Reply:
Thanks, looks good. I am fine with `null` values passed to `set` being removed from the model. As you said, there shouldn't be a place where a `null` value needs to be supported.